### PR TITLE
statistics: some refactoring about stats collector

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve?

Some refactoring

### What is changed and how it works?

Rename the current collector to `ReservoirSampleCollector` and introduce the base collector.
To introduce more new collectors in the future.

How it Works:

### Check List 

Tests 

- No code

### Release note 
```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -78,6 +78,7 @@ require (
 	google.golang.org/grpc v1.27.1
 	gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c // indirect
 	gopkg.in/yaml.v2 v2.4.0
+	honnef.co/go/tools v0.2.0 // indirect
```
Comment (qw4990) [inline go.mod:81]: Should this line be removed?

**Review by qw4990 — APPROVED** [review]

**Review by time-and-fate — APPROVED** [review]
LGTM. But an irrelevant test case failed.

**Pull request merged**